### PR TITLE
Issue #13100: Create tests from FileTabCharacter xdoc examples 

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #13100

### CODE REVIEW

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example4.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+FileTabCharacter
+eachLine = (default)false
+fileExtensions = java, xml
+
+
+*/
```
Comment (nrmancuso) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example4.xml:7]: While the html config "comment" is debatable (I am not a big fan), this is not valid xml. As we demand compilation for java files, we should have valid xml. Are we able to simply put this comment inside of a valid multiline xml comment? If fix is that easy, we should also update html file.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example4.xml (hunk 1 above)
Comment (stoyanK7) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example4.xml:7]: Me too, but this is the only way for now. I tried putting the comment as such
```diff
+<!--
/*
FileTabCharacter
eachLine = (default)false
fileExtensions = java, xml


*/
+-->
```

but the `InlineConfigParser` is strict about it
> Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Config not specified on top.Please see other inputs for examples of what is required.


https://github.com/checkstyle/checkstyle/blob/79d3c5536760d2645b0676c5a7f31da3bb0eed72/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L164-L167

Maybe we can create an issue to extend and allow others?

Comment (nrmancuso) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example4.xml:7]: Please do, and take a few minutes to scope work required to allow us to parse config comment successfully with inner Java comment and without (only module config in xml comment).

Comment (stoyanK7) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example4.xml:7]: Opened at https://github.com/checkstyle/checkstyle/issues/13222

Comment (stoyanK7) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example4.xml]: I will remove the commit when #13223 gets merged. I continue with adding other tests.

Comment (stoyanK7) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example4.xml:7]: @nrmancuso I see that the PR/Issue might drag for some time. Let's leave it as is and we can come back to this issue at some point.

Comment (nrmancuso) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example4.xml:7]: Agreed.

**Review by nrmancuso — APPROVED** [review]

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**